src/hooks/useImageHandler.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ChangeEvent } from 'react';
+import { IFormInfo } from 'pages/petition/post';
+
+const useImageHandler = () => {
+   const [imageUrls, setImageUrls] = React.useState<string[]>([]);
+
+   const addImage = (e: ChangeEvent<HTMLInputElement>, formInfo: IFormInfo, setFormInfo: React.Dispatch<React.SetStateAction<IFormInfo>>) => {
+      const imageList: FileList | null = e.target.files;
+      if (imageList) {
+         const fileList = [...formInfo.files];
+         const urlList = [...imageUrls];
+         for (let i = 0; i < imageList.length; i++) {
+            fileList.push(imageList[i]);
+            const imageUrl = URL.createObjectURL(imageList[i]);
+            urlList.push(imageUrl);
+         }
```

Comment:
```suggestion
         const fileList = [...formInfo.files, ...imageList];
         const urlList = [...imageUrls, ...imageList.map(URL.createObjectURL)];
```
`for 문`없이 표현 가능하겠네요! 명령형 프로그래밍과 선언형 프로그래밍에 대해 알아보시면 좋을 것 같습니다...!!